### PR TITLE
Add a edit_site function (#144)

### PR DESCRIPTION
**Pull request by Bvr4**
# Pull Request

## Description

This PR refers to the issue #144 : Create an edit_site function

Creation of the PVSiteEditMetadata pydantic model, with all Optional field, so you can update one or as many field as you want.
Creation of the edit_site function, and tests for it.

## How Has This Been Tested?

I tested via pytest, with different configuration : update a single field, more fields, set a field to None, etc. And made sure that unfilled field are not impacted.

- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pvsite_datamodel/write/user_and_site.py [hunk 1]
```diff
@@ -245,6 +246,45 @@ def update_user_site_group(session: Session, email: str, site_group_name: str) -
...
+        - longitude: longitude of site as an integer
+        - inverter_capacity_kw: inverter capacity of site in kw
+        - module_capacity_kw: module capacity of site in kw
+        - capacity_kw: capacity of site in kw
+        - dno: dno of site
+        - gsp: gsp of site
+    """
+    site = session.query(SiteSQL).filter(SiteSQL.site_uuid == site_uuid).first()
+
+    update_data = site_info.model_dump(exclude_unset=True, exclude_none=False)
```
Comment (peterdudfield) [inline pvsite_datamodel/write/user_and_site.py:273]: nice, so does, drop all the unset values, but keep them if they have been set to None?

Comment (Bvr4) [reply pvsite_datamodel/write/user_and_site.py:273]: Exactly.

**Review by peterdudfield — APPROVED** [review]

**Comment by Bvr4** [comment]
Hello, sorry I just realized I forgot to put the name of the function I created in the README file...

**Comment by peterdudfield** [comment]
> Hello, sorry I just realized I forgot to put the name of the function I created in the README file...

Oh yea, do you mind adding this?

**Pull request merged**